profile/src/main/java/com/scottlogic/deg/profile/reader/atomic/ConstraintValueValidator.java

```diff
@@ -0,0 +1,188 @@
+            .forEach(val->validateAny(type, val));
+    }
+
+    private static void validateTypes(Object value) {
+        OfTypeConstraintFactory.create(new Field("validation"), (String)value);
+    }
+
+    private static void validatePattern(Object value) {
+        try {
+            Pattern.compile((String)value);
+        } catch (IllegalArgumentException e) {
+            throw new InvalidProfileException(e.getMessage());
+        }
+    }
+
+    private static void validateStringLengthInt(AtomicConstraintType type, Object value) {
+        if (!(value instanceof Number)){
+            throw new ValidationException(
+                String.format("Couldn't recognise 'value' property, it must be an Integer but was a %s with value `%s`",
+                    value.getClass().getSimpleName(), value));
+        }
+
+        BigDecimal valueAsBigDecimal = NumberUtils.coerceToBigDecimal(value);
+        if (valueAsBigDecimal.stripTrailingZeros().scale() > 0) {
+            throw new ValidationException(String.format(
+                "Couldn't recognise 'value' property, it must be an integer but was a decimal with value `%s`",
+                value
+            ));
+        }
+        ensureValueBetween(type, value, BigDecimal.ZERO, BigDecimal.valueOf(Defaults.MAX_STRING_LENGTH));
+    }
+
+    private static void validateNumber(AtomicConstraintType type, Object value) {
+        if (!(value instanceof Number)){
+            throw new ValidationException(
+                String.format("Couldn't recognise 'value' property, it must be an Number but was a %s with value `%s`",
+                    value.getClass().getSimpleName(), value));
+        }
+
+        ensureValueBetween(type, value, Defaults.NUMERIC_MIN, Defaults.NUMERIC_MAX);
+
+    }
+
+    private static void validateDateTime(Object value) {
+        if (!(value instanceof OffsetDateTime)){
+            throw new ValidationException(String.format("Dates should be expressed in object format e.g. { \"date\": \"yyyy-MM-ddTHH:mm:ss.SSS[Z]\" }", value));
+        }
+        OffsetDateTime offsetDateTime = (OffsetDateTime) value;
+
+        if (offsetDateTime != null && (offsetDateTime.getYear() > 9999 || offsetDateTime.getYear() < 1)) {
+            throw new ValidationException("Dates must be between 0001-01-01T00:00:00.000Z and 9999-12-31T23:59:59.999Z");
+        }
+    }
+
+    private static void validateGranularity(Object value) {
+        if (value instanceof Number) {
+            ParsedGranularity.parse(value);
+        }
+        else if (value instanceof String) {
+            ParsedDateGranularity.parse((String) value);
+        }
+        else {
+            throw new ValidationException("Couldn't recognise granularity value, it must be either a negative power of ten or one of the supported datetime units.");
+        }
+    }
+
+    private static void ensureValueBetween(AtomicConstraintType type, Object value, BigDecimal min, BigDecimal max) {
```

Comment:
all callers pass in Number value.  So we should tighten up method param so its a Number